src/backend/d3d12/CommandBufferD3D12.cpp

```diff
@@ -258,20 +259,74 @@ namespace d3d12 {
                         BeginRenderPassCmd* beginRenderPassCmd = commands.NextCommand<BeginRenderPassCmd>();
                         currentRenderPass = ToBackend(beginRenderPassCmd->renderPass.Get());
                         currentFramebuffer = ToBackend(beginRenderPassCmd->framebuffer.Get());
+                        currentSubpass = 0;
 
                         uint32_t width = currentFramebuffer->GetWidth();
                         uint32_t height = currentFramebuffer->GetHeight();
                         D3D12_VIEWPORT viewport = { 0.f, 0.f, static_cast<float>(width), static_cast<float>(height), 0.f, 1.f };
                         D3D12_RECT scissorRect = { 0, 0, static_cast<long>(width), static_cast<long>(height) };
                         commandList->RSSetViewports(1, &viewport);
                         commandList->RSSetScissorRects(1, &scissorRect);
-                        commandList->OMSetRenderTargets(1, &device->GetCurrentRenderTargetDescriptor(), FALSE, nullptr);
                     }
                     break;
 
                 case Command::BeginRenderSubpass:
                     {
                         commands.NextCommand<BeginRenderSubpassCmd>();
+                        const auto& subpassInfo = currentRenderPass->GetSubpassInfo(currentSubpass);
+
+                        uint32_t attachmentCount = currentRenderPass->GetAttachmentCount();
+                        bool usingBackbuffer = false; // HACK(enga@google.com): workaround for not having depth attachments
+
+                        DescriptorHeapHandle rtvHeap = device->GetDescriptorHeapAllocator()->AllocateCPUHeap(D3D12_DESCRIPTOR_HEAP_TYPE_RTV, attachmentCount);
+                        uint32_t rtvIndex = 0;
+
+                        for (uint32_t index : IterateBitSet(subpassInfo.colorAttachmentsSet)) {
+                            uint32_t attachment = subpassInfo.colorAttachments[index];
+                            const auto& attachmentInfo = currentRenderPass->GetAttachmentInfo(attachment);
+
+                            ComPtr<ID3D12Resource> texture;
+                            if (auto textureView = currentFramebuffer->GetTextureView(attachment)) {
+                                texture = ToBackend(textureView->GetTexture())->GetD3D12Resource();
+                            } else {
+                                usingBackbuffer = true;
+                                texture = device->GetCurrentTexture();
+                            }
+
+                            D3D12_CPU_DESCRIPTOR_HANDLE rtvHandle = rtvHeap.GetCPUHandle(rtvIndex++);
+                            D3D12_RENDER_TARGET_VIEW_DESC rtvDesc;
```

Comment:
nit: this should be created from the view as it will eventually have information about the mipSlice and planeSlice. Something like a ```TextureView::GetRTVDesc```, no need to cache it in the texture view though.